WordPress/src/main/java/org/wordpress/android/ui/pages/PagesFragment.kt

```diff
@@ -249,6 +264,44 @@ class PagesFragment : Fragment() {
         viewModel.start(site)
     }
 
+    private fun showToast(toastMessageHolder: ToastMessageHolder) {
+        context?.let {
+            toastMessageHolder.show(it)
+        }
+    }
+
+    private fun previewPage(activity: FragmentActivity, page: PageModel) {
+        val post = postStore.getPostByLocalPostId(page.pageId)
+        val action = PreviewPost(
```

Comment:
I'd consider trying to move most of this code into the VM. It'll require some changes, but it would make the code much cleaner and testable. Wdyt?

Reply:
I'm not sure about that one. This is the equivalent code of the handlePostListAction in the post list activity. Ideally we would share the same "coreVM" with both page and post list, but it's out of scope for this patch IMO.